handler.js

```diff
@@ -0,0 +1,115 @@
+import { fileURLToPath } from 'url';
+import path from 'path';
+import fsPromises from 'fs/promises';
+import xml2js from 'xml2js';
+import AWS from 'aws-sdk';
+import 'dotenv/config';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const s3 = new AWS.S3();
+const BUCKET = process.env.BUCKET_NAME;
+
+export const xml_editor = async (event, context) => {
+    const { httpMethod, path: requestPath, headers, body, isBase64Encoded } = event;
+
+
+    if (httpMethod === 'GET' && requestPath === '/') {
+        try {
+            const html = await fsPromises.readFile(path.join(__dirname, 'index.html'), 'utf8');
+            return {
+                statusCode: 200,
+                headers: {
+                    'Content-Type': 'text/html',
+                },
+                body: html,
+            };
+        } catch (error) {
+            console.error('Error reading HTML:', error);
+            return {
+                statusCode: 500,
+                body: JSON.stringify({ message: 'Internal server error' }),
+            };
+        }
+    }
+
+
+    if (httpMethod === 'GET' && requestPath === '/generate-presigned-url') {
```

Comment:
don't do this you can have multiple lamdas why you are writing the if for url and method ?